src/targets/gpu/mlir.cpp

```diff
@@ -589,8 +591,52 @@ std::string dump_mlir(const module& m)
     return mlir_print(&mlirOperationPrint, mod_op);
 }
 
-code_object_op compile_mlir(const context&, const module& m)
+void adjust_param_shapes(module& m, const std::vector<instruction_ref>& inputs)
 {
+    auto names = m.get_parameter_names();
+    std::sort(names.begin(), names.end());
+    for(auto i : range(names.size()))
+    {
+        const auto& name  = names[i];
+        const auto& input = inputs[i]->get_shape();
+        auto param        = m.get_parameter(name);
+        if(input.standard())
+            continue;
+        auto lens = input.lens();
+        std::vector<operation> ops;
+        if(input.transposed())
+        {
+            auto perm  = find_permutation(input);
+            auto iperm = invert_permutation(perm);
+            lens       = reorder_dims(lens, iperm);
+            ops.push_back(make_op("transpose", {{"permutation", perm}}));
+        }
+        if(input.broadcasted())
+        {
+            std::transform(lens.begin(),
+                           lens.end(),
+                           input.strides().begin(),
```

Comment:
If input is transposed, don't we need to reorder strides as well?

Reply:
I mean, 
when it first goes into
  if(input.transposed())

lens is reordered as the given permutation.
And then if it goes into 
  if(input.broadcasted())

In std::transform, input.strides() needs to be reordered as well.

I'm not sure I understood correctly but please check this.

Reply:
You are right this needs to be fixed.